include/internal/provider.h

```diff
@@ -87,6 +87,8 @@ int ossl_provider_get_capabilities(const OSSL_PROVIDER *prov,
                                    OSSL_CALLBACK *cb,
                                    void *arg);
 int ossl_provider_self_test(const OSSL_PROVIDER *prov);
+int ossl_provider_random(const OSSL_PROVIDER *prov, int which, void *buf, size_t n,
```

Comment:
Naming - I'd have "get" or something else in the name - e.g. ossl_provider_get_random - or alternatively ossl_provider_get_random_bytes as it stands out as a little odd in the existing interface otherwise.

This is basically RAND_bytes_ex and RAND_priv_bytes_ex combined into a provider ...

Reply:
Renamed to `ossl_provider_random_bytes` which matches the RAND calls.